.github/workflows/cmake_find_package.yml

```diff
@@ -0,0 +1,32 @@
+name: cmake_find_package
+
+on:
+    push:
+    pull_request:
+        branches:
+            - main
+
+jobs:
+  cmake-find-package:
```

Comment:
is this testing just `find_package` specifically, or the entire process of installing, building, and testing GSL?

Reply:
1.) It first tests installation.
2.) Tests the find_package support.
3.) It then builds/runs the tests to ensure the installation is 100% valid.

Part 3 is generally overkill but still helpful I think.

Also `actions` unlike `azure-pipelines` can be used in forks which can be helpful when working on a PR and not wanting to publish the PR until you are 100% ready.

Reply:
In that case, can this be renamed to something more general than `cmake_find_package.yaml`? This test seems to be performing an end-to-end test of installing GSL as a standalone project and running tests. Then

The comment `# cmake_find_package.yml explicitly tests find_package support used by package managers` inside `tests/CMakeLists.txt` emphasizes that this test is "explicitly" testing the `find_package` call but that just seems to be a by-product of the installation step.

Also out of curiosity, will replacing generator with something other than "Ninja" test anything different?

Reply:
(linking above comment to review)

Reply:
I've revised the script to just test `find_package` functionality.

> Also out of curiosity, will replacing generator with something other than "Ninja" test anything different?

For find_package testing I don't see a benefit.